src/main/java/org/terasology/simpleFarming/systems/PlantAuthoritySystem.java

```diff
@@ -105,13 +101,7 @@ private boolean isValidPosition(Vector3i position) {
             return false;
         }
         Block targetBlock = worldProvider.getBlock(position);
-        if (targetBlock != airBlock) {
-            return false;
-        }
-        Block belowBlock = worldProvider.getBlock(position.addY(-1));
-        if (belowBlock.isPenetrable()) {
-            return false;
-        }
-        return true;
+        Block belowBlock = worldProvider.getBlock(new Vector3i(position).addY(-1));
```

Comment:
Would wish to avoid the creation of a new Vector3i here as the sheer number of them being created across modules is our biggest drain.

Reply:
Really?  That's sort of depressing.  I'll take it out when I have some time (probably later today), but if that's the case, I'd argue that we need to figure out what the root cause is and fix it, because not being able to create temporary objects in ordinary (non-hotspot) code is going to cause the code to become needlessly complicated, and that's a big problem.

On a related note, I was already frustrated with the lack of support for `ImmutableVector3i` and was thinking of poking that part of the codebase sometime soonish and improving that support.  If creating objects is a problem, though, that implies that improved support for immutables is hopeless and/or pointless.  What are your thoughts here?

Reply:
Gotcha.  I found some discussion on Slack about that.  I'm still not convinced that code like this is the problem (since this code is only going to run once per planted seed); it seems much more likely that other code that runs much more frequently also uses `Vector3i` heavily, and that code needs to be optimized.  Although, I suppose this same code would logically be used by automated planting machines, once a module introduces those, and we might run into the same problem then.  At any rate, could you point me towards the profiler you use for Terasology so I can look into the matter for myself?

At any rate, I'm happy to give your argument the benefit of the doubt and pull out the extra instance.  I also added a comment to make sure that anyone reading the code understands what's going on.